Enhance accessiblity of richcombo label

Comandeer opened the pull request:
<!--
🚨 If you want to submit a PR for a security vulnerability, please contact us directly
at https://ckeditor.com/contact/ instead. 🚨
-->
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches that change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## Did you follow the CKEditor 4 code style guide?

Your code should follow the guidelines from the [CKEditor 4 code style guide](https://github.com/ckeditor/ckeditor4/blob/major/dev/docs/codestyle.md) which helps keep the entire codebase consistent.

- [x] PR is consistent with the code style guide

## What is the proposed changelog entry for this pull request?

```
* [#4493](https://github.com/ckeditor/ckeditor4/issues/4493): Fixed: [Rich Combo](https://ckeditor.com/cke4/addon/richcombo) label does not reflects the current value of the combobox.
```

## What changes did you make?

I've introduced a new element containing the current value of the combobox and linked it alongside the label in combobox's `[aria-labelledby]` attribute.

## Which issues does your PR resolve?

Closes #4493.

sculpt0r commented on plugins/richcombo/plugin.js:
```diff
@@ -318,6 +319,14 @@ CKEDITOR.plugins.add( 'richcombo', {
 				}
+
+				var valueLabel = typeof text != 'undefined' ? text : value,
+					valueLabelElement = this.document.getById( 'cke_' + this.id + '_value_label' ),
```
There are now three places that use `_value_label`. From my point of view, it could be useful to create some helper function which returns a properly formatted id for this element. OTOH, it may end with a big amount of functions like this. However, to change behaviour that relays on template values you don't have to search and alter so many places.

Could we add this kind of 'getter' here?

sculpt0r commented on plugins/richcombo/plugin.js:
```diff
@@ -45,6 +45,7 @@ CKEDITOR.plugins.add( 'richcombo', {
 		' onfocus="return CKEDITOR.tools.callFunction({focusFn},event);"' +
 		' onclick="' + specialClickHandler + 'CKEDITOR.tools.callFunction({clickFn},this);return false;">' +
 			'<span id="{id}_text" class="cke_combo_text cke_combo_inlinelabel">{label}</span>' +
+			'<span id="{id}_value_label" class="cke_combo_label"></span>' +
```
I suppose there is a reason why the currently selected value needs to be in a separate label? Is it some a11y `good practice`? If not, maybe just updating the initial label element could be a better idea? It takes less code in a template.

Comandeer replied:
Actually the first idea was to add `_text` element as a second element referenced by `[aria-labelledby]` – as it would be much easier than introducing logic for changing `_label` element. Unfortunately, `_text` element contains the initial label if user does not select anything. Due to that I introduced `_value_label` element, creating even the more convoluted solution than the logic for changing `_label` element :see\_no\_evil:.

However I'm thinking about even different approach – using both `[aria-labelledby]` and `[aria-describedby]` to [cover more assistive technology than only screen readers](https://www.sarasoueidan.com/blog/accessible-text-labels/). Yet it needs some testing.

sculpt0r replied:
I assume we stay with `[aria-labelledby]` only :)

sculpt0r commented on tests/plugins/richcombo/manual/arialabel.md:
```diff
@@ -0,0 +1,28 @@
+**Note:** This is the test for screen readers, so open one before testing.
+
+1. Press <kbd>Tab</kbd> key multiple times to move focus to the editor.
+2. Press <kbd>Alt+F10</kbd> to switch focus to the toolbar and navigate to the `Styles` combo.
```
Since there is only one combo in the toolbar. We don't need additional navigation in this step. Simple `Alt+F10` keys press put us in a right place.

Comandeer replied:
That's true for tests run on dev version of the editor, but on built version there will be more buttons. That's why I included more verbose instructions.

sculpt0r reviewed (changes requested):
The initial issue is about not updating the label element which is still not updating. However it is all about screen readers, so your solution seems to be fine. Especially that I tested it with NVDA & JAWS on FF and it looks (`sounds`) good to me.

However, could you take a look at my other comments? After those explanations, I think it will be good to merge.

Comandeer commented:
I've modified the whole logic. Now the original label element is modified to include the value of the combo.

I've also modified the order of label to "\<selected value>, \<combo name>" instead of "\<combo name> \<selected value>".

sculpt0r commented on tests/plugins/richcombo/richcombo.js:
```diff
@@ -89,6 +93,62 @@ bender.test( {
+	},
+
+	// (#4493)
+	'test richcombo label after returning to defaul value': function() {
```
I would improve this test case name in the same way as the previous one. This one could be something like `test richcombo label contains default label after richcombo returs to default value`

sculpt0r commented on tests/plugins/richcombo/richcombo.js:
```diff
@@ -89,6 +93,62 @@ bender.test( {
+	},
+
+	// (#4493)
+	'test richcombo initial label': function() {
```
Could we improve this test case name, so that it will express what we are expecting? I mean something like: `test created richcombo label contains initial label value`

sculpt0r reviewed (changes requested):
I've run unit tests for Edge, All IE, FF, Chrome, Safari, and all passed - so :+1:
 
My JAWS with FF is passing the manual test - also: :+1: 

I have only some suggestions for tests cases names. After that - I think we will be good to go :)

Comandeer commented:
I've updated tests names.

sculpt0r commented:
LGTM :tada:
